dev/com.ibm.ws.microprofile.health/src/com/ibm/ws/microprofile/health/internal/HealthCheckHttpResponseBuilder.java

```diff
@@ -72,19 +73,20 @@ public void addResponses(Set<HealthCheckResponse> hcResponseSet) {
 
     public void setHttpResponse(HttpServletResponse httpResponse) {
         httpResponse.setHeader(HTTP_HEADER_CONTENT_TYPE, MEDIA_TYPE_APPLICATION_JSON);
-        HashMap<String, Object> payload = new HashMap<String, Object>();
+        JsonObjectBuilder payload = jsonBuilderFactory.createObjectBuilder();
 
         // No health check procedure found
-        if (checks.isEmpty()) {
+
+        if (numOfChecks == 0) {
```

Comment:
What was wrong with the old approach that we need this new numOfChecks variable?